Reword the readme and split contributor information into another file

ninjabear opened the pull request:
#159 was the original source of this, but I remembered finding it quite hard to understand the readme before because it contains a lot of information on various aspects of nickel development, imo all it really needs to have is a status, how to helloworld and where to find out more. 

I appreciate the readme is pretty subjective and am not suggesting this PR be accepted entirely as is - I'm open to changing it up

Ryman commented on README.md:
````diff
@@ -1,64 +1,31 @@
-```not_rust
 [dependencies.nickel]
 version = "*"
 features = ["nightly"]
````
This is only required for nightly, and it's "unstable" now instead of "nightly" :sob:

Ryman commented on README.md:
```diff
@@ -1,64 +1,31 @@
 features = ["nightly"]
+
+[dependencies]
+rustc-serialize = "*"
```
`rustc-serialize` was required for the old readme example before I trimmed it down, it's now only needed for some of the examples. It's probably worth adding some sort of explanation within the examples themselves (possibly comments beside the imports) or some comment similar to 'you can find the crates required to build the examples on crates.io'. Not sure which is best!

Ryman commented on README.md:
```diff
@@ -78,20 +45,22 @@ fn main() {
-Nickel is open source and licensed with the [MIT license](https://github.com/nickel-org/nickel/blob/master/LICENSE)
+##License
 
+[nickel.rs](http://nickel.rs) is open source and licensed with the [MIT license](https://github.com/nickel-org/nickel/blob/master/LICENSE)
```
Can probably drop most of the line pre- "[MIT](..)"

Ryman commented:
This looks like a good improvement :+1: 

I've been meaning to get round to this for a while, my preference would be to cut it down even a bit more (I'm kind of [inspired by this recently](https://speakerdeck.com/rowanmanning/ux-for-your-node-modules)). I do think we should slip the example code above the installation instructions for sure!

ninjabear commented:
I really like the readme proposed in that link too - its kind of where I was going with this but I think I've missed some bits out and given some things slightly odd names (Usage->hello world). I can change it to be more like that if you like @Ryman ?

Ryman commented:
My preference would be to put the code sample where dependencies are right now, without any titles (its implied) and then put dependencies below as 'Installation' or similar. But other than that I think most of this PR is worth taking as-is to iterate on :+1: 

@cburgdorf any opinion on this?

ninjabear commented:
@Ryman I went through your comments and implemented, think I caught them all? 

I changed "nightly" to "unstable" but I'm not sure if you think we should remove the channel altogether?

Ryman commented:
Looks good, much appreciated @ninjabear!

ninjabear commented:
np :)
